sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -373,66 +456,63 @@ object CreateDataSourceTableUtils extends Logging {
         schema = relation.schema.map { f =>
           CatalogColumn(f.name, f.dataType.catalogString)
         },
+        partitionColumnNames = relation.partitionSchema.map(_.name),
         properties = tableProperties.toMap,
         viewText = None)
     }
 
-    // TODO: Support persisting partitioned data source relations in Hive compatible format
     val qualifiedTableName = tableIdent.quotedString
     val skipHiveMetadata = options.getOrElse("skipHiveMetadata", "false").toBoolean
-    val (hiveCompatibleTable, logMessage) = (maybeSerDe, dataSource.resolveRelation()) match {
+    val (hiveTable, hiveParts, logMessage) = (maybeSerDe, dataSource.resolveRelation()) match {
```

Comment:
This piece of code is really messy now, it will be good if we can simplify it a little bit.